docs/examples/move/nft_marketplace/sources/rental_extension.move

```diff
@@ -0,0 +1,417 @@
+    public fun list<T: key + store>(
+        kiosk: &mut Kiosk,
+        cap: &KioskOwnerCap,
+        protected_tp: &ProtectedTP<T>,
+        item_id: ID,
+        duration: u64,
+        price_per_day: u64,
+        ctx: &mut TxContext,
+    ) {
+        assert!(kiosk_extension::is_installed<Rental>(kiosk), EExtensionNotInstalled);
+
+        kiosk.set_owner(cap, ctx);
+        kiosk.list<T>(cap, item_id, 0);
+
+        let coin = coin::zero<IOTA>(ctx);
+        let (object, request) = kiosk.purchase<T>(item_id, coin);
+
+        let (_item, _paid, _from) = protected_tp.transfer_policy.confirm_request(request);
+
+        let rentable = Rentable {
+            object,
+            duration,
+            start_date: option::none<u64>(),
+            price_per_day,
+            kiosk_id: object::id(kiosk),
+        };
+
+        place_in_bag<T, Listed>(kiosk, Listed { id: item_id }, rentable);
+    }
+
+    /// Allows the renter to delist an item, that is not currently being rented.
+    /// Places (or locks, if a lock rule is present) the object back to owner's Kiosk.
+    /// Creators should mint an empty TransferPolicy even if they don't want to apply any royalties.
+    /// If they wish at some point to enforce royalties, they can update the existing TransferPolicy.
+    public fun delist<T: key + store>(
+        kiosk: &mut Kiosk,
+        cap: &KioskOwnerCap,
+        transfer_policy: &TransferPolicy<T>,
+        item_id: ID,
+        _ctx: &mut TxContext,
+    ) {
+        assert!(kiosk_extension::is_installed<Rental>(kiosk), EExtensionNotInstalled);
+        assert!(kiosk.has_access(cap), ENotOwner);
+
+        let rentable = take_from_bag<T, Listed>(kiosk, Listed { id: item_id });
+
+        let Rentable {
+            object,
+            duration: _,
+            start_date: _,
+            price_per_day: _,
+            kiosk_id: _,
+        } = rentable;
+
+        if (has_rule<T, LockRule>(transfer_policy)) {
+            kiosk.lock(cap, transfer_policy, object);
+        } else {
+            kiosk.place(cap, object);
+        };
+    }
+
+    /// This enables individuals to rent a listed Rentable.
+    ///
+    /// It permits anyone to borrow an item on behalf of another user, provided they have the
+    /// Marketplace extension installed.
+    ///
+    /// The Rental Policy defines the portion of the coin that will be retained as fees and added to
+    /// the Rental Policy's balance.
+    public fun rent<T: key + store>(
+        renter_kiosk: &mut Kiosk,
+        borrower_kiosk: &mut Kiosk,
+        rental_policy: &mut RentalPolicy<T>,
+        item_id: ID,
+        mut coin: Coin<IOTA>,
+        clock: &Clock,
+        ctx: &mut TxContext,
+    ) {
+        assert!(kiosk_extension::is_installed<Rental>(borrower_kiosk), EExtensionNotInstalled);
+
+        let mut rentable = take_from_bag<T, Listed>(renter_kiosk, Listed { id: item_id });
+
+        let max_price_per_day = MAX_VALUE_U64 / rentable.duration;
```

Comment:
It could be just checked when the rentable is listed in `list()`

Reply:
Done